### PR TITLE
Campaign Copy, Strategy Copy, Strategy Bulk Copy Functionality

### PR DESCRIPTION
**Pull request by jitendra-xoriant**
Campaign Copy, Strategy Copy, Strategy Bulk Copy Functionality

### CODE REVIEW

File: src/main/java/com/mediamath/terminalone/TerminalOne.java [hunk 1]
```diff
@@ -361,6 +361,26 @@ public Strategy save(Strategy entity) throws ClientException, ParseException {
...
+	 * @param entity
+	 *            expects Strategy entity.
+	 * @return Strategy object.
+	 * @throws ClientException
+	 *             a client exception is thrown if any error occurs.
+	 * @throws ParseException
+	 *             a parse exception is thrown when the response cannot be
+	 *             parsed.
+	 */
+	public JsonResponse<? extends T1Entity> BulkCopy(Strategy entity) throws ClientException, ParseException {
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/TerminalOne.java]: Can we make methods camelCase?

File: src/main/java/com/mediamath/terminalone/models/BulkStrategy.java [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+ * limitations under the License.
+ ******************************************************************************/
+
+package com.mediamath.terminalone.models;
+
+public class BulkStrategy {
+	
+	private int id;
+	private String name;
+	private boolean  supply_management;
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/models/BulkStrategy.java]: Are these the only fields that can be copied in a strategy?

File: src/main/java/com/mediamath/terminalone/models/Campaign.java [hunk 3]
```diff
@@ -134,6 +134,7 @@
 	private boolean use_mm_freq;
 	private int version;
 	private String zone_name;
+	private boolean copyCampaign=false;
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/models/Campaign.java]: Why do we need this flag?

File: src/main/java/com/mediamath/terminalone/models/Strategy.java [hunk 4]
```diff
@@ -709,6 +713,38 @@ public void setAggregate(Aggregate aggregate) {
 		this.aggregate = aggregate;
 	}
 
+	public boolean isCopyStrategy() {
```
Comment (karthik-mediamath) [inline src/main/java/com/mediamath/terminalone/models/Strategy.java]: Is this to differentiate between strategy create and strategy copy?

File: src/main/java/com/mediamath/terminalone/TerminalOne.java (hunk 1 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/TerminalOne.java]: yes.

File: src/main/java/com/mediamath/terminalone/models/BulkStrategy.java (hunk 2 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/models/BulkStrategy.java]: no we can not, as we need some object which we can put in list for request items to be formed as 
strategies.1.id
strategies.2.id

File: src/main/java/com/mediamath/terminalone/models/Campaign.java (hunk 3 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/models/Campaign.java]: to form a URL with /copy.

if the falg is set then its copy funtionality

File: src/main/java/com/mediamath/terminalone/models/Strategy.java (hunk 4 above)
Comment (jitendra-xoriant) [inline src/main/java/com/mediamath/terminalone/models/Strategy.java]: not likely, flag is only set to understand that we need to add /copy to URL

**Pull request merged**